### DISCUSSION TITLE
Frigate+ labeling: What to do with low res objects?

### DISCUSSION THREAD

**Discussion by nuansed** [blakeblackshear/frigate#3536] [Ask A Question] [answered]
A 4K camera can pick up objects for recognition that are quite far away. The objects are easily recognisable in the original resolution image. 

Frigate+ downscales the displayed image to 640 pixels wide for labelling. This makes it difficult to identify and label far away objects in the downscaled image: 
<img width="435" alt="Screenshot 2022-07-26 at 9 38 39 PM" src="https://user-images.githubusercontent.com/12543435/181020760-bf21c260-d340-45e7-9b81-3c309f786b75.png">
<img width="170" alt="Screenshot 2022-07-26 at 9 46 46 PM" src="https://user-images.githubusercontent.com/12543435/181021652-8b580560-fe46-4859-8dc1-0848785a9d6f.png">
<img width="174" alt="Screenshot 2022-07-26 at 9 48 52 PM" src="https://user-images.githubusercontent.com/12543435/181022115-15ab3727-f6c2-457b-bff3-94ba4f0f9ef9.png">

While labelling, I'm occasionally not sure if some of the pixels that stand out are indeed an object to be labeled or not. How should I proceed in that case? Leave the potential object unlabelled? Or label it if I'm more than 50% confident? 

Setting the bounding box exactly to the objects borders also becomes more inaccurate. I tend to err towards a larger box. Does a slightly inaccurate box affect training?

**Comment by NickM-27** [comment]
@blakeblackshear Will likely have a better answer, but from what I understand the way you've labeled it seems to work as you said it is stored in high resolution in the backend. Also if the bounding box on the higher resolution is still too small then the object can be ignored in training. 

As far as box sizing goes, I think a few pixels of extra space won't matter but as the example images show you don't want to have too much extra space.

**Comment by blakeblackshear** [comment, marked as answer]
Just do your best. I am going to add an option to see a higher resolution image when labeling in the next set of updates.
